packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts

```diff
@@ -31,58 +28,17 @@ export abstract class CrossDomainCommunicator {
     return undefined;
   }
 
-  postMessage<M extends Message>(
-    params: MessageWithOptionalId<M>,
-    options?: { bypassTargetOriginCheck: boolean }
-  ) {
+  async postMessage(message: Message, options?: { bypassTargetOriginCheck: boolean }) {
+    await this.connect();
```

Comment:
this change should have no effect. callers of this class used to be responsible to call it. review the code changes below.